src/gateways/createHighlightClient.ts

```diff
@@ -8,19 +9,23 @@ const formatError = (response: Response) => {
     });
 };
 
-export default (url: string) => {
+export default (
+  basePath: string,
+  getAuthorizedFetchConfig: () => Promise<FetchConfig> = () => Promise.resolve({credentials: 'include'})
+) => {
   const config = new Configuration({
-    basePath: url,
-    credentials: 'include',
-    fetchApi: (...args) => fetch(...args)
-      .then((response) => {
-        if (response.status === 422) {
-          return formatError(response);
-        } else if (response.status === 401) {
-          return Promise.reject(new UnauthenticatedError());
-        }
-        return Promise.resolve(response);
-      }),
+    basePath,
+    fetchApi: async(url: string, fetchConfig: FetchConfig) => {
+      const authorizedFetchConfig = await getAuthorizedFetchConfig();
+      const response = await fetch(url, {...fetchConfig, ...authorizedFetchConfig});
```

Comment:
this'll prevent the given `fetchConfig` from specifying headers sometimes, you may want to use one of the deep merge functions either from lodash or ts-utils